content/reference/stacks.md

```diff
@@ -0,0 +1,195 @@
+---
+title: "Stacks"
+icon: "fa fa-layer-group"
+description: "Stacks are a way to express the complete infrastructure needed for an environment using a standard YAML configuration"
+weight: 1
+tags:
+- yaml
+- atmos
+- configuration
+- concept
+---
+
+# Background
+
+Stacks are a central piece of SweetOps. They're a set of YAML files that follow a standard schema that enable a **fully declarative description of your various environments**. This empowers you with the ability to separate your infrastructure's environment configuration settings from the business logic behind it (provided via components).
+
+SweetOps utilizes YAML configuration for stacks because it's an easy to work with format that is nicely portable across multiple tools. The stack yaml format is natively supported today via [atmos](https://github.com/cloudposse/atmos), [the terraform-yaml-stack-config module](https://github.com/cloudposse/terraform-yaml-stack-config), and [Spacelift](https://spacelift.io) via [the terraform-spacelift-cloud-infrastructure-automation module](https://github.com/cloudposse/terraform-spacelift-cloud-infrastructure-automation).
+
+# Conventions
+
+We have a number of important conventions around stacks that are worth noting.
+
+## Stack Files
+
+Stack files can be very numerous in large cloud environments (think many dozens to hundreds of stack files). To enable proper organization of stack files, SweetOps recommends the following:
+
+* All stacks should be stored in a `stacks/` folder at the root of your infrastructure repository.
+* Global values you that you want to share across all stacks should go into a `stacks/globals.yaml` file
+  * This file name is special in that it will not be treated as a stack by any SweetOps automation tooling.
+* Name individual environment stacks as `$env-$stage.yaml`
+  * For example, `$env` could be `ue2` (for `us-east-2`) and `$stage` might be `prod` which would result in `stacks/ue2-prod.yaml`
+* When you have configuration that should be shared across `env`'s, you should use `gbl` as your `env` and then share that configuration using `import`.
```

Comment:
One thing I'm not understanding properly: Is there some reason that we utilize `gbl` for global files like `gbl-security.yaml`, but we utilize `globals` for other global config files like `uw2-globals.yaml` and `globals.yaml`? 

Shouldn't all "import only, don't apply" configuration files like these be using the same `globals` pattern so they can get filtered out by this code?:

```hcl
  config_filenames   = fileset("../../../stacks", "*.yaml")
  stack_config_files = [for f in local.config_filenames : f if(replace(f, "globals", "") == f)]
```

Reply:
`gbl` is a convention that relates to "global" resources. E.g. in AWS Route53 and IAM are global and not regional, while an RDS database is regional.

The `globals` file relates to global imports, not a global region.

Reply:
Gotcha. My confusion here stem'd from the fact that I **thought** we `import`'d `glb-*` files from the root `stacks/` folder in at least one customer's infrastructure, but now I can't seem to find that... so nevermind. Must have misread something 👍